source/client.c

```diff
@@ -69,6 +72,8 @@ static void s_mqtt_client_shutdown(
     (void)bootstrap;
     (void)channel;
 
+    AWS_LOGF_TRACE(AWS_LS_MQTT_CLIENT, "Channel has been shutdown with error code %d", error_code);
```

Comment:
I'd move this forward a line and add the connection state to the log message since so much logic is conditionally executed on it.  Might be able to remove some of the below messages that are essentially just state indicators too if you switched this to DEBUG.

Reply:
I like the more specific debug messages, and this general trace, but I'll add the connection id.